proposals/Exceptions.md

```diff
@@ -262,130 +249,475 @@ Like the `block`, `loop`, and `if` instructions, the `try` instruction is a
+try block of the catch block, since instructions in the body of the catch block
+are not in the body of the try block.
+
+Once a catching try block is found for the thrown exception, the operand stack
+is popped back to the size the operand stack had when the try block was entered,
+and then an except_ref referring to the caught exception is pushed back onto the
+operand stack.
+
+If control is transferred to the body of a catch block, and the last instruction
+in the body is executed, control then exits the try block.
+
+If the selected catch block does not throw an exception, it must yield the
+value(s) expected by the corresponding catching try block. This includes popping
+the caught exception.
+
+Note that a caught exception can be rethrown using the `rethrow` instruction.
+
+### Rethrowing an exception
+
+The `rethrow` instruction takes the exception associated with the `except_ref`
+on top of the stack, and rethrows the exception. A rethrow has the same effect
+as a throw, other than an exception is not created. Rather, the referenced
+exception on top of the stack is popped and then thrown.
+
+### Exception data extraction
+
+The `if_except` block begins with an `if_except` instruction, and has two
+instruction blocks,
+
+That is, the forms of an if_except block is:
+
+```
+if_except block_type except_index
+  Instruction*
+end
+
+if_except block_type except_index
+  Instruction*
+else
+  Instruction*
+end
+```
+
+In the first form, the instructions between the `if_except` and 'end' define the
+`then block`. In the second form, the instructions between the `if_except` and
+`else` define the `then block`, while the instructions between the `else` and
+the `end` define the `else block`.
+
+The conditional query of an exception checks the exception tag of exception on
+top of the stack. It succeeds only if the exception index of the instruction
+matches the corresponding exception tag. Once the query completes, the exception
+is popped off the stack.
+
+If the query succeeds the values (associated with the popped exception) are
+extracted and pushed onto the stack, and control transfers to the instructions
+in the then block.
+
+If the query fails, it either enters the else block, or transfer control to the
+end of the if_except block if there is no else block.
+
+### Stack traces
+
+When an exception is thrown, the runtime will pop the stack across function
+calls until a corresponding, enclosing try block is found. It may also associate
+a stack trace that can be used to report uncaught exceptions. However, the
+details of this is left to the embedder.
+
+## Changes to the text format.
+
+This section describes change in the [instruction syntax
+document](https://github.com/WebAssembly/spec/blob/master/document/core/instructions.rst).
+
+### New instructions
+
+The following rules are added to *instructions*:
+
+```
+  try resulttype instruction* catch instruction* end |
+  except except_index |
```

Comment:
The "except except_index" line confused me on my read-through the first time around. Is this still accurate or can we drop this line?

Reply:
I think we can drop this line. Thanks for the catch.